Elements/src/Serialization/JSON/JsonInheritanceConverter.cs

```diff
@@ -144,17 +144,17 @@ public override void WriteJson(Newtonsoft.Json.JsonWriter writer, object value,
 
                 // Operate on all identifiable Elements with a path less than Entities.xxxxx
                 // This will get all properties.
-                if (value is Element && writer.Path.Split('.').Length == 1 && !ElementwiseSerialization)
+                if (value is Element element && writer.Path.Split('.').Length == 1 && !ElementwiseSerialization)
                 {
-                    var ident = (Element)value;
+                    var ident = element;
                     writer.WriteValue(ident.Id);
                 }
                 else
                 {
                     var jObject = Newtonsoft.Json.Linq.JObject.FromObject(value, serializer);
-                    if (jObject.TryGetValue(_discriminator, out _))
+                    if (jObject.TryGetValue(_discriminator, out JToken token))
                     {
-                        jObject[_discriminator] = GetDiscriminatorName(value);
+                        ((JProperty)token).Value = GetDiscriminatorName(value);
```

Comment:
We can avoid a lookup by using the value provided to us by the method. Before we were just throwing it away.